Cannot print volatile variables: error: invalid static_cast from type const volatile unsigned int* to type const void*

**`volatile_cast_test.cpp`**
```c++
#include "tinyformat.h"

int main(int argc, char const *argv[])
{
    volatile unsigned int wait_time = 10;
    std::cerr << tfm::format( "Volatile '%s'\n", wait_time ) << std::endl;
    return 0;
}
```

**`g++ -o main volatile_cast_test.cpp && ./main`**
```
In file included from volatile_cast_test.cpp:2:0:
tinyformat.h: In instantiation of ‘tinyformat::detail::FormatArg::FormatArg(const T&) [with T = volatile unsigned int]’:
tinyformat.h:966:32:   required from ‘tinyformat::detail::FormatListN<N>::FormatListN(const Args& ...) [with Args = {volatile unsigned int}; int N = 1]’
tinyformat.h:1019:20:   required from ‘tinyformat::detail::FormatListN<sizeof... (Args)> tinyformat::makeFormatList(const Args& ...) [with Args = {volatile unsigned int}]’
tinyformat.h:1055:37:   required from ‘void tinyformat::format(std::ostream&, const char*, const Args& ...) [with Args = {volatile unsigned int}; std::ostream = std::basic_ostream<char>]’
tinyformat.h:1064:11:   required from ‘std::__cxx11::string tinyformat::format(const char*, const Args& ...) [with Args = {volatile unsigned int}; std::__cxx11::string = std::__cxx11::basic_string<char>]’
volatile_cast_test.cpp:7:60:   required from here
tinyformat.h:509:23: error: invalid static_cast from type ‘const volatile unsigned int*’ to type ‘const void*’
             : m_value(static_cast<const void*>(&value)),
```

--- a/tinyformat.h
+++ b/tinyformat.h
@@ -506,7 +506,9 @@ class FormatArg
 
         template<typename T>
         FormatArg(const T& value)
-            : m_value(static_cast<const void*>(&value)),
+            // C-style cast here allows us to also remove volatile; we put it
+            // back in the *Impl functions before dereferencing to avoid UB.
+            : m_value((const void*)(&value)),
             m_formatImpl(&formatImpl<T>),
             m_toIntImpl(&toIntImpl<T>)
         { }

--- a/tinyformat_test.cpp
+++ b/tinyformat_test.cpp
@@ -286,6 +286,9 @@ int unitTests()
 
     //------------------------------------------------------------
     // Misc
+    volatile int i = 1234;
+    CHECK_EQUAL(tfm::format("%d", i), "1234");
+
 #ifdef TEST_WCHAR_T_COMPILE
     // Test wchar_t handling - should fail to compile!
     tfm::format("%ls", L"blah");

